### PR TITLE
csound: restore opcodes

### PR DESCRIPTION
**Pull request by nwhetsell**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR restores Csound functionality that was removed in #79186. This is accomplished by using a resource block with a GitHub repository commit hash. I’m not sure if that’s acceptable (although `brew audit --strict csound` is fine with it).

### CODE REVIEW

File: Formula/csound.rb [hunk 1]
```diff
@@ -79,6 +93,23 @@ def install
...
+    resource("csound-plugins").stage do
+      mkdir "build" do
+        system "cmake", "..",
+                        "-DABLETON_LINK_HOME=#{buildpath}/ableton-link",
+                        "-DBUILD_ABLETON_LINK_OPCODES=ON",
+                        "-DBUILD_CHUA_OPCODES=OFF",
+                        "-DCSOUNDLIB=CsoundLib64",
+                        "-DCSOUND_INCLUDE_DIR=#{include}/csound",
+                        "-DCS_FRAMEWORK_DEST=#{frameworks}",
+                        "-DUSE_FLTK=OFF",
```
Comment (carlocab) [inline Formula/csound.rb:108]: Do we need this flag even if we already have an `fltk` dependency? (Or does changing it to `ON` introduce additional ones?)

Comment (nwhetsell) [reply Formula/csound.rb:108]: > Do we need this flag even if we already have an `fltk` dependency?

Sort of. Csound’s [FLTK widgets](https://csound.com/docs/manual/ControlFltkIntro.html) are *currently* being built with the rest of Csound, using code in https://github.com/csound/csound, at

https://github.com/Homebrew/homebrew-core/blob/5c3ed83e3af4e6abdae77a0a3dc0e38d5d448766/Formula/csound.rb#L70

What the flag does is prevent FLTK widgets from being built a second time using code in https://github.com/csound/plugins. If we want to build Csound’s FLTK widgets using code in https://github.com/csound/plugins, we can remove that flag, but we’d need to add it in

https://github.com/Homebrew/homebrew-core/blob/5c3ed83e3af4e6abdae77a0a3dc0e38d5d448766/Formula/csound.rb#L58-L67

Either way, if the situation with Csound’s FLTK widgets ever changes, it’ll be caught in the test (`FLrun`):

https://github.com/Homebrew/homebrew-core/blob/5c3ed83e3af4e6abdae77a0a3dc0e38d5d448766/Formula/csound.rb#L97-L100

Comment (carlocab) [reply Formula/csound.rb:108]: Ok, that's fine then.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**